augur/filter.py

```diff
@@ -1050,7 +1059,20 @@ def get_groups_for_subsampling(strains, metadata, group_by=None):
                     })
                 # month = (year, month)
                 metadata['month'] = list(zip(metadata['year'], metadata['month']))
-            # TODO: support group by day
+            if 'week' in generated_columns_requested:
+                # skip ambiguous days
+                df_skip = metadata[metadata['day'].isnull()]
+                metadata.dropna(subset=['day'], inplace=True)
+                for strain in df_skip.index:
+                    skipped_strains.append({
+                        "strain": strain,
+                        "filter": "skip_group_by_with_ambiguous_day",
```

Comment:
Ah, this should probably be added to `report_template_by_filter_name`? 
Maybe we should also add this `--group-by week` to the tests in `subsample-skip-ambiguous-dates.t`?

Reply:
Good catch, and yes that would make sense. I'll do that.

I'm also thinking to:

1. Warn when `year` is also included with `week`. It will already be included implicitly, but with ISO year.
2. Error when `month` is also included with `week`. This "works" but the time bins would be irregular since months and weeks don't line up. I don't see any practical use for it. Wouldn't make sense to warn because it's unclear if the user wants `month` or `week`.

Reply:
Ah, both the warning and the error make sense! That also made me realize we should probably check for ambiguous year and month when grouping by week.

Reply:
All done in [force-push](https://github.com/nextstrain/augur/compare/c0abbdab211053022fa727bd3afbb8d6d70af0f9..f2a6d7182eb86e46ab806f962d7bd30ed3dc18ec).

Reply:
> we should probably check for ambiguous year and month when grouping by week.

Oh right... that should also apply to `--group-by year month`, which is only checking for ambiguous month right now. I'll look into that.

Reply:
@joverlee521 I created a new issue #1071 for the above comment, and addressed it in a separate PR.